x-pack/plugins/canvas/public/components/var_config/var_config.tsx

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState, FC } from 'react';
+import {
+  EuiAccordion,
+  EuiButtonIcon,
+  EuiToken,
+  EuiToolTip,
+  EuiText,
+  EuiInMemoryTable,
+  EuiSpacer,
+  EuiButton,
+} from '@elastic/eui';
+
+import { CanvasVariable } from '../../../types';
+import { ComponentStrings } from '../../../i18n';
+
+import { EditVar } from './edit_var';
+import { DeleteVar } from './delete_var';
+
+import './var_config.scss';
+
+const { VarConfig: strings } = ComponentStrings;
+
+enum PanelMode {
+  List,
+  Edit,
+  Delete,
+}
+
+const typeToToken = {
+  number: 'tokenNumber',
+  boolean: 'tokenBoolean',
+  string: 'tokenString',
+};
+
+interface Props {
+  variables: CanvasVariable[];
+
+  onCopyVar: (v: CanvasVariable) => void;
+  onDeleteVar: (v: CanvasVariable) => void;
+  onAddVar: (v: CanvasVariable) => void;
+  onEditVar: (oldVar: CanvasVariable, newVar: CanvasVariable) => void;
+}
+
+export const VarConfig: FC<Props> = ({
+  variables,
+  onCopyVar,
+  onDeleteVar,
+  onAddVar,
+  onEditVar,
+}) => {
+  const [panelMode, setPanelMode] = useState<PanelMode>(PanelMode.List);
+  const [selectedVar, setSelectedVar] = useState<CanvasVariable | null>(null);
+
+  const selectAndEditVar = (v: CanvasVariable) => {
+    setSelectedVar(v);
+    setPanelMode(PanelMode.Edit);
+  };
+
+  const selectAndDeleteVar = (v: CanvasVariable) => {
+    setSelectedVar(v);
+    setPanelMode(PanelMode.Delete);
+  };
+
+  const actions = [
+    {
+      name: strings.getCopyActionButtonLabel(),
+      description: strings.getCopyActionTooltipLabel(),
+      icon: 'copyClipboard',
+      type: 'icon',
+      isPrimary: true,
+      onClick: onCopyVar,
+    },
```

Comment:
Is there anywhere else in Kibana that has this behavior? I'm not super certain I know of a good way to do this without writing a new component instead of using the existing helper component from EUI

Reply:
No worries about this one. It's not absolutely necessary and was just a nice-to-have on my part. I was originally thinking we might be able to pull off some fanciness with swapping the `icon` key's value based on some time-based state change during the `onClick` function, but I'm probably oversimplifying it or just flat-out wrong...lol :)